docs/methodology/timeseries_imputation.rst

```diff
@@ -0,0 +1,162 @@
+BA-level data to impute the BA timeseries, and only subregion data to impute the
+subregion data, or we could bundle all of the timeseries together and impute them in a
+single run, with the BAs informing the subregion results and vice-versa.
+
+It turns out that because the algorithm is good at selecting which of the individual
+timeseries within the input data are most appropriate for imputing a given set of
+missing values, these two approaches provide almost identical results. We've chosen to
+bundle all of the EIA-930 timeseries together in a single imputation run because it
+simplifies resource management in the PUDL ETL -- the linear algebra operations that it
+relies on are heavily parallelized internally, and doing all of the imputation in a
+single job means that we don't have to worry about more than one imputation running
+simultaneously resulting in contention for CPU resources.
+
+For simplicity, the FERC-714 data is imputed in a separate run, but in the future we
+might consider combining the FERC-714 and EIA-930 data into a single imputation run,
+since they both report the same underlying variable and should have very similar curves.
+
+After the imputation is complete, we split out the imputed BA and subregion data into
+distinct tables, since they are each associated with different additional columns.
+
+Evaluating Imputation Performance
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We use a form of hold-out or simulation-based validation to measure the performance of
+our imputation. We artificially remove reported values from months of data that have no
+values flagged for imputation using real patterns of missignness observed in months
+where many values were flagged for imputation. Then we impute the missing values in a
+validation pipeline that is separate from the main ETL pipeline, and calculate an error
+metric comparing the originally reported values to the imputed values. The error metric
+we're using is the Mean Absolute Percentage Error (MAPE) via
+:func:`sklearn.metrics.mean_absolute_percentage_error`.
+
+In outline the validation pipeline works as follows:
+
+1. Identify "bad" months where many values were imputed (a month in this case is
+   specific to a single reporting organization).
+2. Identify "good" months with no imputed values (from any reporting organization).
+3. Match one "good" month with one "bad" month.
+4. Use the pattern of flagged values from the "bad" month to null values in the "good"
+   month, flagging them as "simulated".
+5. Impute any "simulated" null values using all the other time series available to
+   inform the imputation.
+6. Compare the imputed and reported values and compute the MAPE.
+7. (optionally, in production) Check that the MAPE is less than a configurable threshold
+   (currently set to 5%) and raise an error if it is not.
+
+This validation pipeline can be enabled in production so it runs every night, or it can
```

Comment:
Is it possible for users to access the data created in validation? Not sure if it's saved as an artifact anywhere or could be uploaded to Zenodo or something. It's great that you already let users see the imputation codes in `out_ferc714__hourly_planning_area_demand`, sharing the validation data would be a nice (but also not necessary if it's a heavy lift) additional step.

Reply:
That would be awesome. Consider it an "above and beyond" sort of thing.